### PR TITLE
docs: Add client requirements template to default README contents

### PR DESCRIPTION
**Pull request by Josmithr**
Also updates `Client Requirements.md` to just use the template directly. Effectively moves the source of truth for that document into `markdown-magic`.

[AB#8577](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/8577)

### CODE REVIEW

File: ClientRequirements.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Platform Requirements for Fluid Framework Client Packages
```
Comment (Josmithr) [inline ClientRequirements.md:1]: For reviewers: this document used to be the "source of truth" of this information. This PR would move that source of truth into the markdown-magic templates directory, but we still want to retain this information at the root of the repo for discoverability.

File: tools/markdown-magic/src/templates/Client-Requirements-Template.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+These are the platform requirements for the current version of Fluid Framework Client Packages.
```
Comment (Josmithr) [inline tools/markdown-magic/src/templates/Client-Requirements-Template.md:1]: For reviewers: this becomes the new "source of truth" for the client requirements documentation.

**Review by kashms — APPROVED** [review]
LGTM

**Review by CraigMacomber — APPROVED** [review]
Output. Seems good to me. I didn't review the code changes much.

**Pull request merged**